src/charm.py

```diff
@@ -477,104 +278,23 @@ def _common_install_code(self, state: CharmState) -> bool:  # noqa: C901
             logger.error("Failed to setup logrotate")
             raise
 
-        if state.instance_type == InstanceType.OPENSTACK:
-            return True
-
-        self.unit.status = MaintenanceStatus("Installing packages")
-        try:
-            # The `_start_services`, `_install_deps` includes retry.
-            self._install_local_lxd_deps()
-            self._start_services(state.charm_config.token, state.proxy_config)
-        except SubprocessError:
-            logger.error("Failed to install or start local LXD runner dependencies")
-            raise
-
-        self._refresh_firewall(state)
-
-        runner_manager = self._get_runner_manager(state)
-        if not runner_manager.has_runner_image():
-            self.unit.status = MaintenanceStatus("Building runner image")
-            runner_manager.build_runner_image()
-        runner_manager.schedule_build_runner_image()
-
-        self._set_reconcile_timer()
-
-        self.unit.status = MaintenanceStatus("Downloading runner binary")
-        try:
-            runner_info = runner_manager.get_latest_runner_bin_url()
-            logger.info(
-                "Downloading %s from: %s", runner_info["filename"], runner_info["download_url"]
-            )
-            self._stored.runner_bin_url = runner_info["download_url"]
-            runner_manager.update_runner_bin(runner_info)
-        # Safe guard against transient unexpected error.
-        except RunnerBinaryError as err:
-            logger.exception("Failed to update runner binary")
-            # Failure to download runner binary is a transient error.
-            # The charm automatically update runner binary on a schedule.
-            self.unit.status = MaintenanceStatus(f"Failed to update runner binary: {err}")
-            return False
-
-        self.unit.status = ActiveStatus()
         return True
 
     @catch_charm_errors
     def _on_install(self, _: InstallEvent) -> None:
         """Handle the installation of charm."""
-        state = self._setup_state()
```

Comment:
This removes the charm state validation (I know, its kind of implicit). It might be OK to go ahead, but just as a heads up, this will change the current behaviour (previously, the charm would block instead of installing for wrong config, for example).

Reply:
Good point. Reviewing the code, the _setup_state is called in both _in_start and in _on_config_changed, so if you believe it is safe (I would say it is), I will not readd it again